core/src/main/java/tc/oc/pgm/filters/MonostableFilter.java

```diff
@@ -0,0 +1,112 @@
+package tc.oc.pgm.filters;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.filter.ReactorFactory;
+import tc.oc.pgm.api.filter.query.MatchQuery;
+import tc.oc.pgm.api.filter.query.Query;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.Tickable;
+import tc.oc.pgm.api.time.Tick;
+import tc.oc.pgm.filters.dynamic.FilterMatchModule;
+import tc.oc.pgm.filters.dynamic.Filterable;
+import tc.oc.pgm.filters.dynamic.Filterables;
+
+public class MonostableFilter extends SingleFilterFunction implements ReactorFactory {
+
+  private final Duration duration;
+  final Class<? extends Filterable<?>> scope;
+
+  // Filterables that currently pass the inner filter, mapped to the instants that they expire.
+  // They are not actually removed until the inner filter goes false.
+  final Map<Filterable<?>, Instant> endTimes = new HashMap<>();
```

Comment:
Filters are map-scoped and should contain no mutable state in them

Checking the original implementation (https://github.com/OvercastNetwork/ProjectAres/blob/master/PGM/src/main/java/tc/oc/pgm/filters/matcher/match/MonostableFilter.java) all state was properly contained within the reactor, which is a match-scoped

Reply:
Yes, this is something I forgot to move over...